themes/dotmatrix/dotmatrix.theme

```diff
@@ -0,0 +1,1053 @@
+    __dotmatrix_activate_hostname
+}
+
+
+_lp_dotmatrix_theme_directory() {
+    lp_terminal_format ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+    __dotmatrix_directory_FS
+    __dotmatrix_directory_path
+}
+
+
+_lp_dotmatrix_theme_prompt() {
+
+    local header inline prompt
+
+    # Arguments: fg, bg, bold, underline, fallback_fg, fallback_bg
+    # Color codes can be any integer in 256 XTerm's system colors in [0–255].
+    # Special color codes:
+    #   -1 : no color
+    #   -2 : previously set color
+    #   -3 : swap fg and bg
+    local lp_terminal_format
+
+    ####################################################################
+    # HEADER (LEFT)
+    ####################################################################
+
+    header=""
+    header+="${_DOTMATRIX_DISPLAY}"
+    header+="${_DOTMATRIX_CONNECTION}"
+    header+="${_DOTMATRIX_USER}"
+    header+="${_DOTMATRIX_MULTIPLEXER}"
+    header+="${_DOTMATRIX_HOSTNAME}"
+    header+="${_DOTMATRIX_FS}"
+    header+="${_DOTMATRIX_PATH} "
+
+    if _lp_dirstack; then
+        __dotmatrix_make_sep_plug ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+        header+="${dotmatrix_sep}"
+        __dotmatrix_make_field $lp_dirstack ${DOTMATRIX_COLOR_LITE[@]+"${DOTMATRIX_COLOR_LITE[@]}"}
+        header+="${dotmatrix_field}"
+    else
+        # Recall a dark format to correctly reset sections history
+        # to match the path's section colors.
+        lp_terminal_format ${DOTMATRIX_COLOR_DARK[@]+"${DOTMATRIX_COLOR_DARK[@]}"}
+    fi
+
+    __dotmatrix_make_sep_last
+    header+="${dotmatrix_sep}"
+
+    ####################################################################
+    # HEADER SIDE (RIGHT)
+    ####################################################################
+
+    __dotmatrix_make_sep_side ""
+    side="${dotmatrix_sep}"
+
+    # battery / load / temperature
+    __dotmatrix_make_sep_plug ${DOTMATRIX_COLOR_DARK[@]+"${DOTMATRIX_COLOR_DARK[@]}"}
+    local blt_plug="${dotmatrix_sep}"
+
+    local battery=""
+    if _lp_battery ; then
+        __dotmatrix_make_field "$lp_battery" ${DOTMATRIX_COLOR_DARK[@]+"${DOTMATRIX_COLOR_DARK[@]}"}
+        battery="${dotmatrix_field}${LP_MARK_BATTERY}"
+    fi
+
+    local load=""
+    if _lp_load ; then
+        __dotmatrix_make_field "$lp_load" ${DOTMATRIX_COLOR_DARK[@]+"${DOTMATRIX_COLOR_DARK[@]}"}
+        load="${dotmatrix_field}${LP_MARK_LOAD}"
+    fi
+
+    local temperature=""
+    if _lp_temperature ; then
+        __dotmatrix_make_field "$lp_temperature" ${DOTMATRIX_COLOR_DARK[@]+"${DOTMATRIX_COLOR_DARK[@]}"}
+        temperature="${dotmatrix_field}${LP_MARK_TEMP}"
+    fi
+
+    __dotmatrix_make_sep_soft ${DOTMATRIX_COLOR_SEP_DARK[@]+"${DOTMATRIX_COLOR_SEP_DARK[@]}"}
+    # Non-bracketed variables are intentionals,
+    # so that if the variable is null, it does not count in the arguments array.
+    __dotmatrix_join_fields "$dotmatrix_sep" $battery $load $temperature
```

Comment:
This is unsafe. You can't rely on `IFS` being anything sane. Quote these things here, and in `__dotmatrix_join_fields()` on line 913, first check if each element is the empty string (`[[ -z $f ]] && continue`).